flaml/ml.py

```diff
@@ -541,8 +541,7 @@ def evaluate_model_CV(
         total_fold_num += 1
         val_loss_folds.append(val_loss_i)
         if log_training_metric or not isinstance(eval_metric, str):
-            if isinstance(metric_i, dict):
-                log_metric_folds.append(metric_i)
+            log_metric_folds.append(metric_i)

```

Comment:
I think this "if" needs to be removed too.

Reply:
I think we can not remove this "if".  @sonichi 
1. If users do not want log_training_metric (set it to False), this "if" could save some computation resources.
2. sometimes, metric dicts are not norm. It can be {"metric1": float, "metric2": list}. [test/automl](https://github.com/microsoft/FLAML/blob/main/test/automl/test_custom_hp.py#:~:text=def%20test_custom_hp_nlp()%3A  ). It will be hard for us to set default_cv_score_agg_func

Reply:
1. Even when log_training_metric is False, `metric_i` is not necessarily None, e.g., containing "pred_time".
2. The case of `"metric2": list` is problematic. @liususan091219 could you propose a solution?

Reply:
@liususan091219 Thanks! It fixed now.